### PR TITLE
GLUE-11151 Add `findCompanyBusinessUnitById` facade method.

### PR DESCRIPTION
**Pull request by AsonUnique**
**Expected version:** 1.6.0

- Developer(s): @AsonUnique 

- Ticket: https://spryker.atlassian.net/browse/GLUE-11151

- Academy PR: -

- Release Group: https://release.spryker.com/release-groups/view/3430

- Release Overview: https://release.spryker.com/release/pull-request/8096

#### Please confirm

- [ ] No new OS components - or they have been approved by the legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   CompanyBusinessUnit   | minor                 |                       |

-----------------------------------------

##### Change log

Improvements

- Introduced `CompanyBusinessUnitFacade::findCompanyBusinessUnitById()` facade method for retrieving `CompanyBusinessUnitTransfer` by id.

### CODE REVIEW

File: src/Spryker/Zed/CompanyBusinessUnit/Persistence/CompanyBusinessUnitRepository.php [hunk 1]
```diff
@@ -162,4 +188,17 @@ protected function filterCompanyBusinessUnitCollection(
             $companyBusinessUnitQuery->filterByIdCompanyBusinessUnit_In($criteriaFilterTransfer->getCompanyBusinessUnitIds());
         }
     }
+
```
Comment (dmiseev) [inline src/Spryker/Zed/CompanyBusinessUnit/Persistence/CompanyBusinessUnitRepository.php:191]: get rid empty line

**Review by dmiseev — APPROVED** [review]

**Review by romanhavrylko — APPROVED** [review]

File: src/Spryker/Zed/CompanyBusinessUnit/Persistence/CompanyBusinessUnitRepository.php [hunk 2]
```diff
@@ -20,6 +20,11 @@
  */
 class CompanyBusinessUnitRepository extends AbstractRepository implements CompanyBusinessUnitRepositoryInterface
 {
+    /**
+     * @var string
+     */
+    protected const TABLE_JOIN_PARENT_BUSINESS_UNIT = 'parentCompanyBusinessUnit';
```
Comment (floriankraemer) [inline src/Spryker/Zed/CompanyBusinessUnit/Persistence/CompanyBusinessUnitRepository.php]:
```suggestion
    protected const TABLE_JOIN_PARENT_COMPANY_BUSINESS_UNIT = 'parentCompanyBusinessUnit';
```
I'm not sure if we have a convention for how to name this constant. But wouldn't this make it more clear?

**Review by floriankraemer — CHANGES_REQUESTED** [review]
Just one possible change, please check my comment.

File: src/Spryker/Zed/CompanyBusinessUnit/Persistence/CompanyBusinessUnitRepository.php (hunk 2 above)
Comment (AsonUnique) [reply src/Spryker/Zed/CompanyBusinessUnit/Persistence/CompanyBusinessUnitRepository.php]: Fixed

**Review by floriankraemer — APPROVED** [review]

**Review by floriankraemer — APPROVED** [review]

**Review by floriankraemer — APPROVED** [review]

**Pull request closed without merging**